students/ericstreit/lessons/lesson06/assignment/good_perf.py

```diff
@@ -0,0 +1,62 @@
+"""
+poorly performing, poorly written module
+
+"""
+
+import datetime
+import csv
+import timeit
+
+def analyze(filename):
+    start = datetime.datetime.now()
+    with open(filename) as csvfile:
+        reader = csv.reader(csvfile, delimiter=',', quotechar='"')
+        new_ones = []
+        found = 0
+        for row in reader:
+            lrow = list(row)
+            if "ao" in lrow[6]:
+                found += 1
+                if lrow[5] > '00/00/2012':
+                    new_ones.append((lrow[5], lrow[0]))
+            else:
+                if lrow[5] > '00/00/2012':
+                    new_ones.append((lrow[5], lrow[0]))
```

Comment:
I would probably have "ao" and the dates be separate if statements, then you don't have duplicate code. Also, I wonder if it takes more time to put new_ones in a list and then put it in the dictionary. Why not just add them directly into the dictionary?